src/main/java/com/amazon/opendistroforelasticsearch/jobscheduler/scheduler/JobScheduler.java

```diff
@@ -145,6 +146,19 @@ boolean reschedule(ScheduledJobParameter jobParameter, JobSchedulingInfo jobInfo
             return true;
         }
         Duration duration = Duration.between(this.clock.instant(), nextExecutionTime);
+
+        Instant secondExecutionTimeFromNow = jobParameter.getSchedule().getNextExecutionTime(nextExecutionTime);
+        if(secondExecutionTimeFromNow != null) {
+            Duration interval = Duration.between(nextExecutionTime, secondExecutionTimeFromNow);
+            if (interval.toMillis() > 0) {
+                double jitter = jobParameter.getJitter() == null ? 0d : jobParameter.getJitter();
+                jitter = jitter > jitterLimit ? jitterLimit : jitter;
```

Comment:
Few questions:
What's reasoning for making jitterLimit configurable?
Why configurable at the node level and not job level?
Why default of 0.95 for jitterLimit?
What's behavior when you have a jitter of 0.95 and a long GC pause that pushes you over a full execution cycle?

Reply:
Thanks, good questions.
1.What's reasoning for making jitterLimit configurable?
We should not allow too big jitter which may be too large to skip next run. For example if the jitter is 1, we may skip next job run. Add maximum jitter limit can protect this.

2.Why configurable at the node level and not job level?
I think node level is enough and add extra parameters in JobParameter will bring extra effort for every job. It's more flexible on job level, but that brings some confusion between "jitterLimit" and "jitter". For one job, if user can clearly define its "jitterLimit", why we need "jitter"?

3.Why default of 0.95 for jitterLimit?
This is to leave some buffer(5%*interval) for next job run. No clear theory for this limit. Considering our job interval is 1 minutes at least, 5% buffer will be 3seconds. The buffer maybe too small for some heavy job, user can configure a smaller jitter limit or set a small jitter on job level.

4.What's behavior when you have a jitter of 0.95 and a long GC pause that pushes you over a full execution cycle?
The long GC pause case is a common problem for any scheduling job no matter what jitter they are using. Long GC pause may skip next job execution cycle. This is a known issue. How about we limit this PR on jitter changes and solve this long GC pause issue separately.

Reply:
"We should not allow too big jitter which may be too large which may skip next run".

Should we reduce the default 0.95 to something lower to reduce the likelihood of this happening?

And yes, GC pause is always a problem, just more curious what the behavior is if it **does** happen? Will it screw anything up in the current behavior/design or will it continue as intended?
Do we skip one execution?

Reply:
Sure, considering 95% is too strict for some long run job, will set it as 60% which will leave 24s for job run.
If GC pause is too long, the next execution may be be postponed which means next execution is skipped.